### PR TITLE
puzzles: deprecate

### PR DESCRIPTION
**Pull request by Aaron-Rumpler**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [ ] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [ ] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

---

The macOS build from upstream hasn't been updated since 2023-09-18. An up-to-date build is available as a formula (see #168128 for context).

### CODE REVIEW

**Comment by khipp** [comment]
From the homepage:

> 2023-09-20: I'm sorry to say that I'm no longer going to be able to build the MacOS version of these puzzles.

File: Casks/p/puzzles.rb [hunk 1]
```diff
@@ -1,27 +1,25 @@
...
-    url "https://www.chiark.greenend.org.uk/~sgtatham/puzzles/devel/"
-    regex(/collection,\s+version\s+(\d+)\.\w+/i)
-    strategy :page_match do |page, regex|
-      # Throttle updates to one every 2 days.
-      next version if Date.parse(version) + 2 > Date.today
-
-      page[regex, 1]
-    end
-  end
+  deprecate! date: "2024-11-16", because: :no_longer_available
```
Comment (khipp) [inline Casks/p/puzzles.rb:10]:
```suggestion
  deprecate! date: "2024-11-16", because: :discontinued
```

Comment (Aaron-Rumpler) [reply Casks/p/puzzles.rb:10]: The software itself is still under development though, and still builds fine for macOS (which [the formula](https://github.com/Homebrew/homebrew-core/blob/master/Formula/p/puzzles.rb) does). Then again, the old version from 2023 is still available from upstream, it's just not updated to the new versions, so `:no_longer_available` might not be exactly suitable either?

Comment (bevanjkay) [reply Casks/p/puzzles.rb:10]: Is the cask the same as the formula (other than the cask being prebuilt)? Maybe we migrate to the formula instead?

Comment (chenrui333) [reply Casks/p/puzzles.rb:10]: yeah, I think they are the same. I would vote to migrate to formula instead

Comment (krehel) [reply Casks/p/puzzles.rb:10]: They are the same, agreed with @chenrui333.  The formula also has 10x the installs of the Cask looking at the 365d period so seems the community is moving in that direction.

**Review by daeho-ro — DISMISSED** [review]

**Comment by Aaron-Rumpler** [comment]
The formula doesn't put the `.app` into `/Applications`. How's a migration going to effect existing users of the cask? I assume it'll remove the cask, meaning that users of the cask will have the `.app` disappear from `/Applications`.

**Comment by SMillerDev** [comment]
Might be better to deprecate this one and point to the formula instead.

**Review by bevanjkay — APPROVED** [review]

**Pull request merged**